controllers/chat/initChatFromPost.js

```diff
@@ -54,39 +54,57 @@ const initChatFromPost = async (req, res) => {
 
     //Add anon info to channel
     if (targetUserModel.is_anonymous) {
-      const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
-        PostAnonUserInfo,
-        User,
-        {
-          postId: postId,
-          userId: targetUserModel.user_id
-        },
-        null,
-        true
-      );
+      let anon_user_info = {};
+      if (source === 'post') {
+        const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
+          PostAnonUserInfo,
+          User,
+          {
+            postId: postId,
+            userId: targetUserModel.user_id
+          },
+          null,
+          true
+        );
+
+        anon_user_info = {
+          anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+          anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+          anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+        };
+      } else if (source === 'comment') {
+        const reaction = await Getstream.feed.getReactionById(commentId);
+        anon_user_info = {
+          anon_user_info_color_code: reaction?.data.anon_user_info_color_code,
+          anon_user_info_color_name: reaction?.data.anon_user_info_color_name,
+          anon_user_info_emoji_code: reaction?.data.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: reaction?.data.anon_user_info_emoji_name
+        };
+      }
 
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code =
-        postAnonUserInfo?.anon_user_info_color_code;
+        anon_user_info?.anon_user_info_color_code;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name =
-        postAnonUserInfo?.anon_user_info_color_name;
+        anon_user_info?.anon_user_info_color_name;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code =
-        postAnonUserInfo?.anon_user_info_emoji_code;
+        anon_user_info?.anon_user_info_emoji_code;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name =
-        postAnonUserInfo?.anon_user_info_emoji_name;
+        anon_user_info?.anon_user_info_emoji_name;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for updating `newStateMemberWithAnonInfo` with anonymous user information is repeated four times. This can be simplified by using a loop or a mapping function. Here's how you could do it:

```diff
- newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code =
-  anon_user_info?.anon_user_info_color_code;
- newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name =
-  anon_user_info?.anon_user_info_color_name;
- newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code =
-  anon_user_info?.anon_user_info_emoji_code;
- newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name =
-  anon_user_info?.anon_user_info_emoji_name;

+ ['color_code', 'color_name', 'emoji_code', 'emoji_name'].forEach((key) => {
+   newStateMemberWithAnonInfo[targetUserModel.user_id][`anon_user_info_${key}`] = anon_user_info[`anon_user_info_${key}`];
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->